src/app/Library/CrudPanel/Traits/Relationships.php

```diff
@@ -132,6 +106,24 @@ protected function changeBelongsToNamesFromRelationshipToForeignKey($data)
         return $data;
     }
 
+    /**
+     * Get the CRUD fields for the current operation, but with
+     * their names ready to be used inside inputs. For example
+     * 'relationship.attribute' => 'relationship[attribute]'.
+     *
+     * @return array
+     */
+    public function fieldsWithOverwrittenNamesForHtml($fields = null)
+    {
+        $fields = $fields ?? $this->fields();
+
+        foreach ($fields as $key => $field) {
+            $fields[$key] = $this->overwriteFieldNameFromDotNotationToArray($field);
+        }
+
+        return $fields;
+    }
+
```

Comment:
Here's a wild thought... any reason not to do all of this... 🥁🥁🥁 inside `CRUD::fields()`? 👀

Since
- before, we were overwriting the field name on addField, the result of `CRUD::fields()` was with them overwritten;
- if `CRUD::fields()` returns the fields with changed names, it would _basically_ return the same thing as before;

That would make this as close as possible to a non-BC, right? The only thing we do that's a BC is that we remove some methods that are no longer useful, and were unfortunately `public`.